src/zod.ts

```diff
@@ -0,0 +1,211 @@
+import { z } from "zod";
+import { Action, LinkType, MatchMode } from "./constants.js";
+import { existsSync } from "fs";
+import { logger } from "./logger.js";
+import { getFileConfig } from "./configuration.js";
+
+const fileConfig = await getFileConfig();
+
+/**
+ * VALIDATION_SCHEMA is just a object of all the zod schemas
+ * each are named after what they are intended to validate
+ */
+
+export const VALIDATION_SCHEMA = z
+	.object({
+		notificationWebhookUrl: z.union([
+			z.undefined(),
+			z
+				.string()
+				.url()
+				.refine((url) => ({
+					message: `invalid notificationWebhookUrl URL - ${url}`,
+				})),
+		]),
+		torznab: z.array(
+			z
+				.string()
+				.url()
+				.refine((url) => ({
+					message: `invalid torznab URL - ${url}`,
+				}))
+		),
+		dataDirs: z.union([
+			z.undefined(),
+			z.array(
+				z.string().refine(
+					(path) => existsSync(path),
+					(path) => ({
+						message: `the dataDirs path ${path} does not exist on the filesystem`,
+					})
+				)
+			),
+		]),
+		torrentDir: z.string().refine(
+			(path) => existsSync(path),
+			(path) => ({
+				message: `the torrentDir path ${path} does not exist on the filesystem`,
+			})
+		),
+		linkDir: z.union([
+			z.undefined(),
+			z.string().refine(
+				(path) => existsSync(path),
+				(path) => ({
+					message: `the linkDir path ${path} does not exist on the filesystem`,
+				})
+			),
+		]),
+		outputDir: z.string().refine(
+			(path) => existsSync(path),
+			(path) => ({
+				message: `the outputDir path ${path} does not exist on the filesystem`,
+			})
+		),
+		includeEpisodes: z.boolean({
+			invalid_type_error:
+				"includeEpisodes must be a defined boolean (true or false)",
+		}),
+		includeSingleEpisodes: z.boolean({
+			invalid_type_error:
+				"includeSingleEpisodes must be a defined boolean (true or false)",
+		}),
+		includeNonVideos: z.boolean({
+			invalid_type_error:
+				"includeNonVideos must be a defined boolean (true or false)",
+		}),
+		duplicateCategories: z.boolean({
+			invalid_type_error:
+				"duplicateCategories must be a defined boolean (true or false)",
+		}),
+		skipRecheck: z.boolean({
+			invalid_type_error:
+				"skipRecheck must be a defined boolean (true or false)",
+		}),
+		maxDataDepth: z.number().gte(1, {
+			message: "maxDataDepth must be a number greater than 0",
+		}),
+		port: z.number().gte(1).lte(65535, {
+			message: "port must be a number between 1 and 65535",
+		}),
+		searchLimit: z.union([z.number(), z.undefined()], {
+			invalid_type_error: "searchLimit must be a number or undefined",
+		}),
+		delay: z.number().gt(1).lt(1000, {
+			message:
+				"delay is in seconds, >1000 implies you think it's milliseconds",
+		}),
```

Comment:
This is 16m 40s, seems like an edge case that it would be set that high. But I guess we shouldn't put constraints on users as long as it meets the type requirements.

I just figured guarding against the chance they think it's milliseconds was prudent.

Reply:
Is there an upper threshold that we should validate against for this situation? 5000 (~1hr20m) or 10000 (~2hr45m) - at what point might we consider this erroneous?

I believe that there is a lower limit that we should enforce (to avoid tracker retaliation mostly) - thinking now, allowing anything under 10s seems like asking for problems.

Thoughts?

Reply:
I think we should just leave it open.